FinniversKit/Sources/Cells/RecentlyFavoritedShelfCell/RecentlyFavoritedShelfCell.swift

```diff
@@ -0,0 +1,275 @@
+import UIKit
+
+public class RecentlyFavoritedShelfCell: UICollectionViewCell {
+    public typealias ButtonAction = ((_ model: RecentlyFavoritedViewmodel, _ isFavorited: Bool) -> ())
+    
+    private static let titleHeight: CGFloat = 8
+    private static let priceTagHeight: CGFloat = 30
+    private let imageviewWidth: CGFloat = 128
+    private var defaultImage: UIImage? {
+        return UIImage(named: .noImage)
+    }
+    
+    public var buttonAction: ButtonAction?
+    private var model: RecentlyFavoritedViewmodel?
+    private var isFavorited: Bool = true {
+        didSet {
+            favoriteButton.isToggled = isFavorited
+        }
+    }
+    
+    private var task: URLSessionDataTask?
+    
+    private lazy var imageView: UIImageView = {
+        let imageView = UIImageView()
+        imageView.translatesAutoresizingMaskIntoConstraints = false
+        imageView.contentMode = .scaleAspectFill
+        imageView.layer.masksToBounds = true
+        imageView.widthAnchor.constraint(equalToConstant: imageviewWidth).isActive = true
+        imageView.heightAnchor.constraint(equalToConstant: imageviewWidth).isActive = true
+        imageView.layer.cornerRadius = 8
+        return imageView
+    }()
+    
+    private lazy var priceLabel: UILabel = {
+        let label = Label(style: .captionStrong)
+        label.textColor = .textTertiary
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.backgroundColor = .clear
+        label.textAlignment = .center
+        
+        return label
+    }()
+    
+    private lazy var priceBackground: UIView = {
+        let background = UIVisualEffectView(withAutoLayout: true)
+        if #available(iOS 13.0, *) {
+            background.effect = UIBlurEffect(style: .systemThinMaterialDark)
+        } else {
+            background.effect = nil
+            background.backgroundColor = UIColor.priceLabelBackgroundColor.withAlphaComponent(0.8)
+        }
+        background.alpha = 1.0
+        background.layer.cornerRadius = RecentlyFavoritedShelfCell.priceTagHeight / 2
+        background.clipsToBounds = true
+        
+        background.contentView.addSubview(priceLabel)
```

Comment:
Could you move this and the setup of the constraints into the `setup()` func? To gather all the view configurations there if possible.

Reply:
will do! 😁